feat(edge): private beta doc (#2206)
scaleway/docs-content: storage/object/reference-content/cname-record.mdx

content:
  h1: CNAME records and DNS for Object Storage with Edge Services
  paragraph: Learn how to create a CNAME record for your customized Edge Services domain, and troubleshoot any potential DNS problems.
tags: object-storage edge-services cname dns
categories: 
  - storage
---

This document contains information to help you successfully create a CNAME record for your customized [Edge Services](/storage/object/how-to/get-started-edge-services) domain, and troubleshoot any potential DNS problems.

## What is a CNAME record?

A **C**anonical **Name** (CNAME) record is a type of [DNS record](/network/domains-and-dns/concepts/#dns-record). Generally, DNS records hold information for translating a domain or subdomain to an IP address, mail server or other domain/subdomain. They are crucial in directing internet traffic to the correct servers. More specifically, CNAME records map one domain name (an alias) to another (the canonical name). 

A CNAME record may look like the following:

| Hostname / Alias           | Destination / Canonical Name  |
|----------------------------|-------------------------------|
| `videos.example.com`       | `otherdomain.com`             |

> In this case, when a DNS server sees this record for `videos.example.com` it will know not to direct traffic to `blog.example.com`'s own IP address, but to that of `otherdomain.com`. It will find `othercomain.com`'s IP address via its [A record](/network/domains-and-dns/concepts/#resource-records).

Reviewer: @RoRoJ good example I am just confused by: 

> it will know not to direct traffic to `blog.example.com`'s own IP address

Did you mean "it will know not to direct traffic to `videos.example.com`'s own IP address" ?

When the client actually connects to `otherdomain.com`'s IP address, the web server can see that the requested URL was `videos.example.com`, and deliver the relevant content.

## When and why do I need to create a CNAME record for Edge Services?

> When you enable Edge Services, initially your bucket content is served through the standard Edge Services endpoint, e.g. `bucket-name.edge.scw.cloud`. If you do not want to customize the standard Edge Services endpoint, you do not need to worry about CNAME records.

Reviewer: @RoRoJ  the edge services' endpoint is now 'bucket-name.svc.edge.scw.cloud'

However, if you choose to [customize your Edge Services endpoint with your own subdomain](/storage/object/how-to/get-started-edge-services/#how-to-configure-a-custom-domain), you are prompted, as part of the process, to create a CNAME record to point your subdomain to the Edge Services endpoint for your bucket. 

    TODO SCREENSHOT

## How to create a CNAME record

Log into your domain provider, and locate the DNS settings for your domain. Create a new CNAME record pointing your subdomain to the Edge Services endpoint for your bucket. This endpoint will be displayed in the Scaleway console.

The interface used by different domain providers varies, but creating your CNAME record may look like one of the following examples:

| Record                                     | Destination                              |
|--------------------------------------------|------------------------------------------|

> | `my-chosen-subdomain.beautiful-domain.com` | `bucket-name.edge.scw.cloud`             |

Reviewer: @RoRoJ the correct target is `bucket-name.svc.edge.scw.cloud.`
I believe you need to add the trailing . (at least in Scaleway DNS product) if you want the DNS to properly interpret the destination - to check in other DNS providers.

| Subdomain                                  | Target host                                    |
|--------------------------------------------|------------------------------------------------|
| `my-chosen-subdomain`                      | `bucket-name.edge.scw.cloud`                   |

| Host record                                | Points to                                      |
|--------------------------------------------|------------------------------------------------|

> | `my-chosen-subdomain`                      | `bucket-name.edge.scw.cloud`                   |

Reviewer: @RoRoJ the correct target is `bucket-name.svc.edge.scw.cloud.`
I believe you need to add the trailing . (at least in Scaleway DNS product) if you want the DNS to properly interpret the destination - to check in other DNS providers.

| `my-chosen-subdomain`                      | Record Type     | Value                                          |
|--------------------------------------------|-----------------|------------------------------------------------|

> | `@`                                        | `CNAME`         | is an alias of `bucket-name.edge.scw.cloud`    |

Reviewer: @RoRoJ the correct target is `bucket-name.svc.edge.scw.cloud.`
I believe you need to add the trailing . (at least in Scaleway DNS product) if you want the DNS to properly interpret the destination - to check in other DNS providers.

You may also see a `TTL`field, which stands for **T**ime **T**o **L**ive.  This tells the DNS resolver how long it can cache this record, before it must re-check the original source in case something has changed. TTL is measured in seconds, and the default value is usually 12 hours (43200 seconds) or 24 hours (86400 seconds). 

 ## Troubleshooting DNS errors

When setting up your customized subdomain with Edge Services, you have the option to carry out a verification check on the CNAME record. Edge Services will query the subdomain and check that it resolves correctly to the Edge Services endpoint. If there is a problem, you will see an error message. 

>     TODO SCREENSHOT

Reviewer 2: NOT TO FORGET

An error message may also display at a later point from your Edge Services dashboard if a problem is detected at any point with your CNAME record or subdomain.

See the table below for help with troubleshooting these errors:

| Error message                             | Solution                                                            |
|-------------------------------------------|---------------------------------------------------------------------|

> | No CNAME record found                     | Make sure you have created a valid CNAME record where your subdomain points to the Edge Services endpoint.  |

Reviewer: @RoRoJ, I believe the issue here is that when we try to resolve the domain name we either realize that there is no record (need to double check with Valentin as no record may be your third case), or the record we found is not a CNAME record (like we get an IP address directly because it's an A or an AAAA record).

> | Incorrect CNAME                           | Make sure your CNAME record points to the Edge Services endpoint in the format `bucket-name.edge.scw.cloud`, and that you have replaced `bucket-name` with the name of your bucket.                                            |

Reviewer: @RoRoJ the correct endpoint is `bucket-name.svc.edge.scw.cloud.` with a trailing . (at least when managed in Scaleway DNS product).

> | Domain does not exist                     | You must own the domain name you are attempting to configure. If you do not already own the domain name, you cannot create a CNAME record for it. Register the domain name, for example using our [Domains and DNS](/network/domains-and-dns/how-to/register-internal-domain/) product, then create a CNAME record for it.                                           |

Reviewer: @RoRoJ, nice idea to suggest using Scaleway DNS.
In this case I believe the issue is that we try to resolve the domain and don't get an answer because nobody registered the domain (no authoritative DNS server was found), or maybe the user made a typo when entering the domain name in the Console.

There are also a number of errors which can occur if you attempt to use an invalid subdomain:

| Error message                             | Solution                                                            |
|-------------------------------------------|---------------------------------------------------------------------|

> | Subdomain error - invalid format          | Make sure the subdomain name you entered is correctly formatted.              |

Reviewer: @RoRoJ, do you think we should add a link to the RFC or other standard document explaining how domain names should be formatted? Or is this overkill?

Author: I can't find anything very user-friendly in the RFC doc, but I've put in a wikipedia link

> | Subdomain error - cannot use `scw.cloud`  | You cannot use `scw.cloud` as your subdomain. Pick another subdomain.          |

Reviewer: @RoRoJ in this case I believe we should say that the user cannot use subdomains of the scw.cloud domain as this domain is owned and managed by Scaleway, and users cannot create records under this domain.

content:
  h1: CNAME records and DNS for Object Storage with Edge Services
  paragraph: Learn how to create a CNAME record for your customized Edge Services domain, and troubleshoot any potential DNS problems.
tags: object-storage edge-services cname dns
categories: 
  - storage
---

This document contains information to help you successfully create a CNAME record for your customized [Edge Services](/storage/object/how-to/get-started-edge-services) domain, and troubleshoot any potential DNS problems.

## What is a CNAME record?

A **C**anonical **Name** (CNAME) record is a type of [DNS record](/network/domains-and-dns/concepts/#dns-record). Generally, DNS records hold information for translating a domain or subdomain to an IP address, mail server or other domain/subdomain. They are crucial in directing internet traffic to the correct servers. More specifically, CNAME records map one domain name (an alias) to another (the canonical name). 

A CNAME record may look like the following:

| Hostname / Alias           | Destination / Canonical Name  |
|----------------------------|-------------------------------|
| `videos.example.com`       | `otherdomain.com`             |

> In this case, when a DNS server sees this record for `videos.example.com` it will know not to direct traffic to `videos.example.com`'s own IP address, but to that of `otherdomain.com`. It will find `othercomain.com`'s IP address via its [A record](/network/domains-and-dns/concepts/#resource-records).

Reviewer 3: ```suggestion
In this case, when a DNS server sees this record for `videos.example.com` it will know not to direct traffic to `videos.example.com`'s own IP address, but to that of `otherdomain.com`. It will find `othercomain.com`'s IP address via its [A record](/network/domains-and-dns/reference-content/understanding-dns-records/#a-record).
```

| Subdomain                                  | Target host                                    |
|--------------------------------------------|------------------------------------------------|
| `my-chosen-subdomain`                      | `bucket-name.svc.edge.scw.cloud.`               |

| Host record                                | Points to                                      |
|--------------------------------------------|------------------------------------------------|
| `my-chosen-subdomain`                      | `bucket-name.svc.edge.scw.cloud.`               |

| `my-chosen-subdomain`                      | Record Type     | Value                                          |
|--------------------------------------------|-----------------|------------------------------------------------|
| `@`                                        | `CNAME`         | is an alias of `bucket-name.svc.edge.scw.cloud.`|

<Message type="tip">
The trailing dot at the end of the target endpoint (`bucket-name.svc.edge.scw.cloud.`) is implicitly added by some domain and DNS providers, and must be explicitly added for others. Check with yours whether the dot is necessary.
</Message>

> You may also see a `TTL`field, which stands for **T**ime **T**o **L**ive.  This tells the DNS resolver how long it can cache this record, before it must re-check the original source in case something has changed. TTL is measured in seconds, and the default value is usually 12 hours (43200 seconds) or 24 hours (86400 seconds). 

Reviewer 3: ```suggestion
You may also see a `TTL` field, which stands for **T**ime **T**o **L**ive.  This tells the DNS resolver how long it can cache this record, before it must re-check the original source in case something has changed. TTL is measured in seconds, and the default value is usually 12 hours (43200 seconds) or 24 hours (86400 seconds). 
```

    <Lightbox src="scaleway-edge-services-cname-error.webp" alt=""/>

An error message may also display at a later point from your Edge Services dashboard if a problem is detected at any point with your CNAME record or subdomain:

    <Lightbox src="scaleway-edge-services-dashboard-error.webp" alt="" />

See the table below for help with troubleshooting these errors:

| Error message                             | Solution                                                            |
|-------------------------------------------|---------------------------------------------------------------------|
| No CNAME record found                     | Make sure you have created a valid DNS record of type **CNAME** (not **A**, **AAAA** or another type), where your subdomain points to the Edge Services endpoint.  |
| Incorrect CNAME                           | Make sure your CNAME record points to the Edge Services endpoint in the format `bucket-name.svc.edge.scw.cloud.`, and that you have replaced `bucket-name` with the name of your bucket.                                            |

> | Domain does not exist                     | You must own the domain name you are attempting to configure. If you do not already own the domain name, you cannot create a subdomain or CNAME record for it. Register the domain name, for example using our [Domains and DNS](/network/domains-and-dns/how-to/register-internal-domain/) product, then create a CNAME record for the subdomain. Otherwise, ensure you did not make a typo when entering the domain name into the Scaleway console                     |

Reviewer 4: ```suggestion
| Domain does not exist                     | You must own the domain name you are attempting to configure. If you do not already own the domain name, you cannot create a subdomain or CNAME record for it. Register the domain name, for example using our [Domains and DNS](/network/domains-and-dns/how-to/register-internal-domain/) product, then create a CNAME record for the subdomain. Otherwise, ensure you did not make a typo when entering the domain name into the Scaleway console.                     |
```
